Amazon.IonObjectMapper.Test/TestObjects.cs

```diff
@@ -234,6 +236,59 @@ public override string ToString()
         }
     }
 
+    public class EngineSerializerFactory : IonSerializerFactory<Engine>
+    {
+        public override IonSerializer<Engine> Create(IonSerializationOptions options, Dictionary<string, object> context)
+        {
+            return new EngineSerializer((CustomSerializerValue)context.GetValueOrDefault("customSerializerKey", null));
+        }
+    }
+
+    public class EngineSerializer : IonSerializer<Engine>
+    {
+        private readonly CustomSerializerValue customSerializerValue;
+        public EngineSerializer(CustomSerializerValue customSerializerValue)
+        {
+            this.customSerializerValue = customSerializerValue;
+        }
+
+        public override Engine Deserialize(IIonReader reader)
+        {
+            return new Engine { Cylinders = customSerializerValue.AddCylinder(reader.IntValue()), ManufactureDate = DateTime.Parse("2009-10-10T13:15:21Z")};
+        }
+
+        public override void Serialize(IIonWriter writer, Engine item)
+        {
+            writer.StepIn(IonType.Struct);
+            writer.SetFieldName("Cylinders");
+            writer.WriteInt(customSerializerValue.RemoveCylinder(item.Cylinders));
+            writer.SetFieldName("ManufactureDate");
+            writer.WriteString(customSerializerValue.ShowCurrentTime());
+            writer.StepOut();
+
+        }
+    }
+
+    public class CustomSerializerValue
```

Comment:
This needs a more expressive name

Reply:
Fixed and moved this test class to a new file.